### PR TITLE
Fix stop logic in FuseIntegrationTest

### PR DESCRIPTION
**Pull request by ddadaal**
### What changes are proposed in this pull request?

Fix the `stop()` logic in FuseIntegrationTest so umounting will work for JNI, JNR and WorkerFuseIntegrationTest.

- For JNI and JNR, umount will happen before cluster stops
- For WorkFuseIntegrationTest, umount will happen after cluster stops

### Why are the changes needed?

This PR fixes the following bug: 

The `stop()` in FuseIntegrationTest currently umounts fuse mountpoint after cluster stops. This works for WorkerIntegrationTest, but for JNI and JNR, umounting will fail with `Input/output error`, since the cluster has already stopped. In JNI and JNR's situations, umount should run before cluster stops.

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for fixing the bug!

**Comment by LuQQiu** [comment]
On-holding merging, workerfuseintegrationtest failed

**Comment by LuQQiu** [comment]
https://github.com/Alluxio/alluxio/runs/5090112308?check_suite_focus=true
Error: 8.170 [ERROR] alluxio.server.worker.WorkerFuseIntegrationTest.touchAndLs  Time elapsed: 17.592 s  <<< ERROR!
ExitCodeException exitCode=1: umount: /tmp/alluxio-tests-fd9a5c51-329d-408b-9ac3-fe64ba0b9f5f/WorkerFuseIntegrationTest-touchAndLs-99ce7716-2c31-420d-a0e2-65a9ae4cbe97: Permission denied

	at alluxio.shell.ShellCommand.run(ShellCommand.java:71)
	at alluxio.util.ShellUtils.execCommand(ShellUtils.java:237)
	at alluxio.client.fuse.AbstractFuseIntegrationTest.stop(AbstractFuseIntegrationTest.java:130)
	at alluxio.client.fuse.AbstractFuseIntegrationTest.after(AbstractFuseIntegrationTest.java:119)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

**Review by ssz1997 — APPROVED** [review]
LGTM. Thanks for fixing this.

**Comment by LuQQiu** [comment]
Error: 0.126 [ERROR] Failures: 
Error: 0.126 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[0]
Error: 0.127 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
Error: 0.127 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
flaky

File: tests/src/test/java/alluxio/server/worker/WorkerFuseIntegrationTest.java
Comment (LuQQiu) [inline tests/src/test/java/alluxio/server/worker/WorkerFuseIntegrationTest.java:43]: worker fuse should be unmounted directly when stopping the worker process.
this line can be added to double check if fuse is unmounted but is not necessary

**Comment by LuQQiu** [comment]
Waiting for the tests and this PR can be merged after validating tests

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**
